htdocs/js/ui/notebook_protection_controller.js

```diff
@@ -0,0 +1,620 @@
+		
+
+		//GROUP TAB METHODS
+		/////////////////////////////////////////
+
+	    $scope.createGroup = function(){
+	    	var pr = prompt("Enter new group name", "");
+
+	    	if (pr != null) {
+
+                GroupsService.createGroup(pr)
+                .then(function(data){
+
+                    $scope.$evalAsync(function(){
+
+                    	$scope.allUsersGroups.push({
+                    		id:data,
+                    		name:pr,
+                    		isAdmin:true
+                    	});
+                    });
+                    console.log('group created '+data);
+                })
+                .catch(function(e){
+                    alert(e);
+                })
+	            
+			}
+	    }
+
+	    $scope.renameGroup = function(){
+	    	var pr = prompt("Enter new group name", $scope.selectedGroup2.name);
+
+	    	if (pr != null) {
+				var r = confirm('Are you sure you want to rename "'+$scope.selectedGroup2.name+' to "'+pr+'"?');
+
+				if(r === true){
+
+					//var groupId;
+	                var prevGroupId = $scope.selectedGroup2.id;
+	                GroupsService.changeGroupName($scope.selectedGroup2.id ,pr)
+	                .then(function(data){
+	                    //groupId = data;
+
+	                    var theIndex;
+						for(var i = 0; i < $scope.allUsersGroups.length; i ++){
+							if($scope.allUsersGroups[i].id === $scope.selectedGroup2.id ){
+								theIndex = i;
+								console.log('at index '+i);
+							}
+						}
+
+	                    $scope.$evalAsync(function(){
+
+	                    	$scope.allUsersGroups[theIndex].name = pr;
+	                    });
+
+	                    //console.log('group changed '+data);
+	                    //alert('group created');
+
+	                })
+	                .catch(function(e){
+	                    alert(e);
+	                })
+
+				}  
+			}
+	    }
+
+	    $scope.deleteGroup = function(){
+	    	var r = confirm('Are you sure you want to delete'+$scope.selectedGroup2.name+'?');
+	    	if(r === true){
+	    		GroupsService.deleteGroup($scope.selectedGroup2.id)
+	    		.then(function(data){
+	    			console.log('deleted '+data);
+
+	    			var theIndex;
+					for(var i = 0; i < $scope.allUsersGroups.length; i ++){
+						if($scope.allUsersGroups[i].id === $scope.selectedGroup2.id ){
+							theIndex = i;
+							//console.log('at index '+i);
+						}
+					}
+
+                    $scope.$evalAsync(function(){
+                    	//remove that item from array
+                    	$scope.allUsersGroups.splice([theIndex], 1);
+                    	//reset the select
+                    	$scope.selectedGroup2 = $scope.allUsersGroups[0];
+                    });
+	    		})
+	    		.catch(function(e){
+                    alert(e);
+                })
+	    	}
+	    }
+
+
+	    //Admins and Memebers areas
+	    ///////////////////////////
+
+
+	    $scope.startWatchingGroups = function(){
+
+	    	var count1 = 0;
+	    	var count2 = 0;
+
```

Comment:
For some reason the callback for watch function is firing right away, even thou they are only supposed to fire when the model changes. So the counts are used to not react to the first callback for watch. Once I figure out why this is happening, I can remove the count variables. Probably has something to do with using $scope.$evalAsync 



> On Jun 10, 2015, at 6:37 PM, Gordon Woodhull <notifications@github.com> wrote:
> 
> In htdocs/js/ui/notebook_protection_controller.js:
> 
> > +	    		.catch(function(e){
> > +                    alert(e);
> > +                })
> > +	    	}
> > +	    }
> > +
> > +
> > +	    //Admins and Memebers areas
> > +	    ///////////////////////////
> > +
> > +
> > +	    $scope.startWatchingGroups = function(){
> > +
> > +	    	var count1 = 0;
> > +	    	var count2 = 0;
> > +
> are these counts just obsolete sanity checks?
> 
> —
> Reply to this email directly or view it on GitHub.
>